source/transfer_list.rst

```diff
@@ -657,5 +658,58 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+
+.. _tpm_evlog_entry:
+
+TPM info table entry layout (XFERLIST_TPM_INFO)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds TPM info of the platform.
+
+.. _tab_tpm_info:
+.. list-table:: TPM info type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **5**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - sizeof (base_address) + sizeof (evlog_size) + sizeof (crb_base_address) + sizeof (crb_size)
+
+   * - evlog_base_address
+     - 8
+     - 8
+     - The base address of event log
+
+   * - evlog_size
+     - 0x4
+     - 16
+     - Size of the event log
```

Comment:
What does the event log contain? Can you provide the file format?

Otherwise this LGTM

Reply:
Typically event log shows information about measurements taken during boot and has values extended to PCR registers. Event log is typically a binary file and follows TCG specifications.

Reply:
@tusharkhandelwal2 wbefore merging this I think we should discuss what we need the EventLog for. 

There are 2 cases that come in mind. 
1. A previous stage boot loader e.g. TF-A constructed the EvenLog but doesn't have a TPM driver.  So the PCRs of the actual hardware were never extended.  In that case a bootloader which has the drivers needed will have to replay the EventLog on the hardware
2. The aforementioned bootloader can extend the PCRs normally, so the only thing we need is the EventLog base addr + size. 

So would it make sense to add a boolean describing if a bootloader needs the replay the EventLog?
We can probably derive that by reading PCR0 (which is required to be extended by the TCG specs), but I am not sure everyone will follow that spec.

Reply:
I would not recommend using a directly link to a specific PDF.  Instead use the more general URL so that if the EFI spec has a minor rev the link will still be valid:
https://trustedcomputinggroup.org/resource/tcg-efi-protocol-specification/

Reply:
> @tusharkhandelwal2 before merging this I think we should discuss what we need the EventLog for.
> 
> There are 2 cases that come in mind.
> 
> 1. A previous stage boot loader e.g. TF-A constructed the EvenLog but doesn't have a TPM driver.  So the PCRs of the actual hardware were never extended.  In that case, a bootloader that has the drivers needed will have to replay the EventLog on the hardware
> 2. The aforementioned bootloader can extend the PCRs normally, so the only thing we need is the EventLog base addr + size.
> 
> So would it make sense to add a boolean describing if a bootloader needs the replay the EventLog? We can probably derive that by reading PCR0 (which is required to be extended by the TCG specs), but I am not sure everyone will follow that spec.

I think adding a "need_to_replay" boolean would be useful.  It should default to zero.